markdown/bitburner.ns.hack.md

```diff
@@ -33,7 +33,7 @@ Function that is used to try and hack servers to steal money and gain hacking ex
 
 A script can hack a server from anywhere. It does not need to be running on the same server to hack that server. For example, you can create a script that hacks the `foodnstuff` server and run that script on any server in the game.
 
-A successful `hack()` on a server will raise that server’s security level by 0.002 per thread. You can use [hackAnalyzeSecurity](./bitburner.ns.hackanalyzesecurity.md) to calculate the security increase for a number of threads.
+A successful `hack()` on a server will raise that server’s security level by 0.002 per "used thread". "used thread" may not be the same as opts.threads. Let's say you call this function with opts.threads = 1000, but it only takes 500 threads to hack all money in the targeted server. In this case, "used thread" is 500. You should use [hackAnalyzeSecurity](./bitburner.ns.hackanalyzesecurity.md) to calculate the security increase.
```

Comment:
```suggestion
A successful `hack()` on a server will raise that server’s security level by 0.002 per "used thread". "Used threads" may not be the same as opts.threads. Let's say you call this function with opts.threads = 1000, but it only takes 500 threads to hack all money in the targeted server. In this case, "used threads" is 500. You should use [hackAnalyzeSecurity](./bitburner.ns.hackanalyzesecurity.md) to calculate the security increase.
```

There is a subtle point with `hack()` that this glosses over: While `grow` is capped based on the number of threads needed to grow to max money, `hack` is capped based on _the number of threads needed to hack from full to 0_. In other words, even if the server has 1 money, you can still "use" up to 500 threads (in your example). There's also an early-out special case for 0 money where used threads will be 0. I.e. the situation is quite complicated for hack.

Reply:
If it's a rule engine (rather than formula) documenting as bullet points makes more sense.